### ISSUE TITLE
`sync_argocd_apps.sh` fails with `argocd` cli version 2.9.0+

### ISSUE THREAD

**Issue by abasu0713** [deployKF/deployKF#70]
### Checks

- [X] I have searched the [existing issues](https://github.com/deployKF/deployKF).
- [X] This issue is NOT specific to the CLI. (If so, please open an issue on the [CLI repo](https://github.com/deployKF/cli/issues))

### deployKF Version

0.1.3

### Kubernetes Version

```shell
Client Version: v1.28.4
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.26.9+k3s1
```

### Description

I was trying to follow the [documentation](https://www.deploykf.org/guides/local-quickstart/)to deploy Kubeflow locally on my Ubuntu 22.04 personal laptop. I run into a blocker [here](https://www.deploykf.org/guides/local-quickstart/#5-sync-argocd-applications) with the following error:
```
$ bash ./deploykf/scripts/sync_argocd_apps.sh

==========================================================================================
Getting admin password from 'Secret/argocd-initial-admin-secret'...
==========================================================================================
>>> DONE

==========================================================================================
Logging in to ArgoCD...
==========================================================================================
FATA[0000] cannot find pod with selector: [app.kubernetes.io/name=] - use the --{component}-name flag in this command or set the environmental variable (Refer to https://argo-cd.readthedocs.io/en/stable/user-guide/environment-variables), to change the Argo CD component name in the CLI 
```
I tried using the ArgoCD Web UI as well to sync the application states. But it didn't work either:
```
$ kubectl get Application -A
NAMESPACE   NAME                                             SYNC STATUS   HEALTH STATUS
argocd      dkf-core--deploykf-auth                          OutOfSync     Missing
argocd      dkf-core--deploykf-dashboard                     OutOfSync     Missing
argocd      dkf-core--deploykf-istio-gateway                 OutOfSync     Missing
argocd      dkf-core--deploykf-profiles-generator            OutOfSync     Healthy
argocd      dkf-dep--cert-manager                            OutOfSync     Missing
argocd      dkf-dep--istio                                   OutOfSync     Missing
argocd      dkf-dep--kyverno                                 OutOfSync     Missing
argocd      dkf-opt--deploykf-minio                          OutOfSync     Missing
argocd      dkf-opt--deploykf-mysql                          OutOfSync     Missing
argocd      kf-dep--argo-workflows                           OutOfSync     Missing
argocd      kf-tools--katib                                  OutOfSync     Missing
argocd      kf-tools--notebooks--jupyter-web-app             OutOfSync     Missing
argocd      kf-tools--notebooks--notebook-controller         OutOfSync     Missing
argocd      kf-tools--pipelines                              OutOfSync     Missing
argocd      kf-tools--poddefaults-webhook                    OutOfSync     Missing
argocd      kf-tools--tensorboards--tensorboard-controller   OutOfSync     Missing
argocd      kf-tools--tensorboards--tensorboards-web-app     OutOfSync     Missing
argocd      kf-tools--training-operator                      OutOfSync     Missing
argocd      kf-tools--volumes--volumes-web-app               OutOfSync     Missing
argocd      deploykf-app-of-apps                             Synced        Healthy
```

No updates to sample yaml was made.

Any suggestions will be much appreciated. 
I am trying to do this to test and understand pipelines without having to spin up a Cloud VM or use a Cloud Kubernetes solution. This is only for development purpose. My laptop is Alienware r17 x1. It has a 3080TI. I wanna use the GPU addons and try local pipelines and hyper-parameter tuning. ANY help in getting me running will be much appreciated. 

Please don't hesitate to reach out in case you have any follow up questions.

### Relevant Logs

```shell
$ bash ./deploykf/scripts/sync_argocd_apps.sh

==========================================================================================
Getting admin password from 'Secret/argocd-initial-admin-secret'...
==========================================================================================
>>> DONE

==========================================================================================
Logging in to ArgoCD...
==========================================================================================
FATA[0000] cannot find pod with selector: [app.kubernetes.io/name=] - use the --{component}-name flag in this command or set the environmental variable (Refer to https://argo-cd.readthedocs.io/en/stable/user-guide/environment-variables), to change the Argo CD component name in the CLI
```

### deployKF Values (Optional)
```shell
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: deploykf-app-of-apps
  namespace: argocd
  labels:
    app.kubernetes.io/name: deploykf-app-of-apps
    app.kubernetes.io/part-of: deploykf
spec:
  project: "default"
  source:
    ## source git repo configuration
    ##  - we use the 'deploykf/deploykf' repo so we can read its 'sample-values.yaml'
    ##    file, but you may use any repo (even one with no files)
    ##
    repoURL: "https://github.com/deployKF/deployKF.git"
    targetRevision: "v0.1.3"
    path: "."

    ## plugin configuration
    ##
    plugin:
      name: "deploykf"
      parameters:

        ## the deployKF generator version
        ##  - available versions: https://github.com/deployKF/deployKF/releases
        ##
        - name: "source_version"
          string: "0.1.3"

        ## paths to values files within the `repoURL` repository
        ##  - the values in these files are merged, with later files taking precedence
        ##  - we strongly recommend using 'sample-values.yaml' as the base of your values
        ##    so you can easily upgrade to newer versions of deployKF
        ##
        - name: "values_files"
          array:
            - "./sample-values.yaml"
  destination:
    server: "https://kubernetes.default.svc"
    namespace: "argocd"
```
_No response_

**Comment by thesuperzapper** [comment]
@abasu0713 thanks for raising, this is a known issue with `argocd` CLI version `2.9.0+` (https://github.com/argoproj/argo-cd/issues/16266).

You will need to __downgrade your `argocd` CLI__ to `2.8.6` or earlier.

## If you are using Homebrew

- Follow my [instructions for installing older `brew` packages](https://github.com/argoproj/argo-cd/issues/16266#issuecomment-1831133650)

## If you are not using Homebrew

- Download the [appropriate `argocd-linux-xxxx` from the `2.8.6` release](https://github.com/argoproj/argo-cd/releases/tag/v2.8.6)

**Comment by abasu0713** [comment]
Thank you for such a quick turnaround. I am unblocked. Appreciate the help!

**Comment by thesuperzapper** [comment]
I have put a temporary warning on the [getting started](https://www.deploykf.org/guides/getting-started/#sync-argocd-applications) and [local quickstart](https://www.deploykf.org/guides/local-quickstart/#5-sync-argocd-applications) pages to remind users to downgrade the ArgoCD CLI to `2.8.6`, but it looks like ArgoCD will be releasing `2.10.0` soon which should resolve this problem.

**Comment by thesuperzapper** [comment]
This bug is resolved in version `2.10.0` of the `argocd` cli.

If you were previously pinning the version in homebrew, you can run:

```bash
brew unpin argocd
brew update
brew upgrade
```
